examples/authentication.ipynb

```diff
@@ -0,0 +1,70 @@
+{
+ "metadata": {
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.8.5-final"
+  },
+  "orig_nbformat": 2,
+  "kernelspec": {
+   "name": "python3",
+   "display_name": "Python 3.8.5 64-bit ('harmony-py': venv)",
+   "metadata": {
+    "interpreter": {
+     "hash": "31862cc71836a94b0e0781803a3648767fc4cb197cc35bade0ddf231ddce7d7c"
+    }
+   }
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 2,
+ "cells": [
+  {
+   "source": [
+    "# Using Harmony-Py Authentication\n",
+    "\n",
+    "In `harmony-py` you instantiate a `Client` in order to communicate with Harmony. To make it easy to store your validated credentials `harmony-py` provides an `authenticate()` function which returns a session object to be passed as an argument to `Client`.\n"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# note: to run, you may want to create a soft link to the library for now: `ln -s ../harmony_py/ harmony_py`\n",
+    "# note: the keyword argument `verify` defaults to True, meaning it will verify the credentials are valid during session instantiation.\n",
+    "\n",
+    "from harmony_py.auth import authenticate\n",
+    "\n",
+    "# specify both username and password\n",
+    "session = authenticate(username='myusername', password='supersecret', verify=False)\n",
+    "\n",
+    "# specify just a username, in which case prompt the user for their password\n",
+    "session = authenticate(username='myusername', verify=False)\n",
+    "\n",
+    "# or look for .netrc credentials\n",
+    "session = authenticate(netrc_file=True, verify=False)\n",
+    "\n",
+    "# Someday, this will work:\n",
```

Comment:
I'm not sure if we should include this, but I'm wondering if the line above should basically be this? Can we just default netrc_file to True and only let users override to say `netrc_file=False`. That way it automatically checks for a  .netrc file or env vars EDL_USERNAME and EDL_PASSWORD when you call `session = authenticate()`?

Reply:
I had wanted to make using a .netrc file a conscious decision of a harmony-py user. It just seemed like bad design to make a no-args 'authenticate()', rely on netrc. Doing it this way, a person must choose the path they wish to take: 1) supply a username and password, or 2) supply a username and be prompted for a password, or 3) explicitly state 'I want to use netrc and I know how to troubleshoot that if it doesn't work'.

We can do the automatic thing though and I can include env behavior.

Reply:
Other people can weigh in. I based my comment on the way the requests Python library works automatically using a .netrc without having to specify to use one.

Reply:
I'm probably 66.6% in favor of defaulting to the netrc file if one exists. But I also agree that explicit is better than implicit, I don't have a strong opinion, but the default to netrc is slightly preferable with the current API. I can think of a couple tweaks to authenticate that might help. I'll comment below.

Reply:
We had a conversation yesterday resulting in this new format:
```
session = authenticate()    # Use ENV variables, then .netrc
session = authenticate(auth=(username, password))
session = authenticate(token='ABCD1234')
```
.netrc use is automatic, ENV use is automatic and happens before .netrc, auth kwarg is used for (username, password) tuple of str, and token is a kwarg to a BearerToken str.